Refactor os deps (#111)
membraneframework/bundlex: README.md

  or a tuple `{:precompiled, <string representing URL to the precompiled library>}` and `lib_name` 
  is a string representing a library that should be provided by a given providers. 
  In case `pkg_config` provider is used, the `pkg-config` will be used to resolve the compilation 
  and linking flags (note that the given libary needs to be previously installed in the system). 
  In case `{:precompiled, <string representing URL to the precompiled library>}` is used, 
  the precompiled library will be first downloaded from the provider URL and appropriate compilation 
  and linking flags will be set. 
  The list of providers is resolved with the order of elements in the providers list - 
  in case one provider fails to provide a library, the next provider from the list is tried.
* `deps` - Dependencies in the form of `{app, lib_name}`, where `app`
is the application name of the dependency, and `lib_name` is the name of lib
specified in Bundlex project of this dependency. Empty list by default. See _Dependencies_ section below
for details.
* `src_base` - Native files should reside in `project_root/c_src/<src_base>`
(application name by default).

> * `compiler_flags` - Custom flags for compiler. Default `-std` flag for `:c` is `-std=c11` and for `:cpp` is `-std=c++17`. 

Reviewer: Description of this option in `typespec` in `Bundlex.Project` is much poorer. Ensure, that `typespec` from `Bundlex.Project` covers infomations previously included in README

> - libs - can be used by other resources as dependencies (via `deps` option)

Reviewer 2: Perhaps we shouldn't mention the `deps` option at all, since we are not describing it anywhere in the README, and instead describe more comprehensively what are these "other resources"? 
```suggestion
- libs - can be used by other resources (that is: other natives and libs) as dependencies
```
WDYT?

> The sources should reside in `project_root/c_src/my_app` directory (this can be changed with `src_base` option).

Reviewer 2: I would make that readme completely "options agnostic" but that's just my opinion :D
```suggestion
By default, the sources should reside in `project_root/c_src/my_app` directory.
```
